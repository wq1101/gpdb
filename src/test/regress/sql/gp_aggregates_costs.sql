set optimizer=off;
set statement_mem="1800";
create table cost_agg_t1(a int, b int, c int);
insert into cost_agg_t1 select i, random() * 99999, i % 2000 from generate_series(1, 1000000) i;
create table cost_agg_t2 as select * from cost_agg_t1 with no data;
insert into cost_agg_t2 select i, random() * 99999, i % 300000 from generate_series(1, 1000000) i;
explain select avg(b) from cost_agg_t1 group by c;
explain select avg(b) from cost_agg_t2 group by c;
insert into cost_agg_t2 select i, random() * 99999,1 from generate_series(1, 200000) i;
analyze cost_agg_t2;
explain select avg(b) from cost_agg_t2 group by c;
drop table cost_agg_t1;
drop table cost_agg_t2;
reset statement_mem;
reset optimizer;
