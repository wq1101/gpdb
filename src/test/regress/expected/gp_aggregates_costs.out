set optimizer=off;
set statement_mem="1800";
create table cost_agg_t1(a int, b int, c int);
insert into cost_agg_t1 select i, random() * 99999, i % 2000 from generate_series(1, 1000000) i;
create table cost_agg_t2 as select * from cost_agg_t1 with no data;
insert into cost_agg_t2 select i, random() * 99999, i % 300000 from generate_series(1, 1000000) i;
explain select avg(b) from cost_agg_t1 group by c;
                                                QUERY PLAN                                                
----------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice2; segments: 3)  (cost=16552.97..16577.97 rows=2000 width=36)
   ->  Finalize HashAggregate  (cost=16552.97..16577.97 rows=667 width=36)
         Group Key: c
         ->  Redistribute Motion 3:3  (slice1; segments: 3)  (cost=16345.00..16525.00 rows=2000 width=36)
               Hash Key: c
               ->  Partial HashAggregate  (cost=16345.00..16405.00 rows=2000 width=36)
                     Group Key: c
                     ->  Seq Scan on cost_agg_t1  (cost=0.00..11345.00 rows=333334 width=8)
 Optimizer: Postgres query optimizer
(9 rows)

explain select avg(b) from cost_agg_t2 group by c;
                                              QUERY PLAN                                               
-------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice2; segments: 3)  (cost=36645.07..40332.35 rows=294982 width=36)
   ->  HashAggregate  (cost=36645.07..40332.35 rows=98328 width=36)
         Group Key: c
         ->  Redistribute Motion 3:3  (slice1; segments: 3)  (cost=0.00..31345.00 rows=333334 width=8)
               Hash Key: c
               ->  Seq Scan on cost_agg_t2  (cost=0.00..11345.00 rows=333334 width=8)
 Optimizer: Postgres query optimizer
(7 rows)

insert into cost_agg_t2 select i, random() * 99999,1 from generate_series(1, 200000) i;
analyze cost_agg_t2;
explain select avg(b) from cost_agg_t2 group by c;
                                              QUERY PLAN                                               
-------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=43614.00..58686.07 rows=463756 width=36)
   ->  HashAggregate  (cost=43614.00..49410.95 rows=154586 width=36)
         Group Key: c
         ->  Redistribute Motion 3:3  (slice2; segments: 3)  (cost=0.00..37614.00 rows=400000 width=8)
               Hash Key: c
               ->  Seq Scan on cost_agg_t2  (cost=0.00..13614.00 rows=400000 width=8)
 Optimizer: Postgres query optimizer
(7 rows)

drop table cost_agg_t1;
drop table cost_agg_t2;
reset statement_mem;
reset optimizer;
